node/forest_libp2p/src/rpc/codec.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use super::{RPCRequest, RPCResponse};
+use bytes::BytesMut;
+use forest_encoding::{error::Error as EncodingError, from_slice, to_vec};
+use futures_codec::{Decoder, Encoder};
+use std::fmt;
+
+/// Codec used for inbound connections. Decodes the inbound message into a RPCRequest, and encodes the RPCResponse to send.
+pub struct InboundCodec;
+/// Codec used for outbound connections. Encodes the outbound message into a RPCRequest to send, and decodes the RPCResponse when received.
+pub struct OutboundCodec;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum RPCError {
+    Codec,
+    Custom(String),
+}
+impl From<std::io::Error> for RPCError {
+    fn from(err: std::io::Error) -> Self {
+        Self::Custom(err.to_string())
+    }
+}
+
+impl From<EncodingError> for RPCError {
+    fn from(_: EncodingError) -> Self {
+        Self::Codec
+    }
+}
```

Comment:
Related to above, doesn't this mean we discard the message from the endcoding error?

Reply:
youre right. fixed. missed that